lib/linguist/languages.yml

```diff
@@ -3047,13 +3047,21 @@ PLSQL:
   tm_scope: none
   color: "#dad8d8"
   extensions:
-  - ".pls"
```

Comment:
> The primary extension should be the first in the list..

Oh, I didn't realise this was the way it worked - thought they all needed to be sorted alphabetically.

> Is `.pls` really no longer considered the primary extension?

I also didn't realise `pls` was considered the primary. I know personally, I never use this extension. Looking up, I see its for `PL/SQL Source`, even so, I didnt think it was that common. If I do a search for `create or replace` alongside that extension, I get 1260 results. Compare for instance to pks (package spec), there are 7 089. Not saying the number of results is a clear indication of what is "primary" and what isn't - just thought it's worth pointing out there are far more instances of an extension that is not `pls`.

In fact, if I open up SQL Developer and go to save a new source file, it defaults onto SQL

![image](https://cloud.githubusercontent.com/assets/1747643/25361340/078779a4-2991-11e7-91b0-a5c42dbd3260.png)

This could just be the fact the SQL Worksheet is typically to run SQL*Plus scripts, that support sql plus directives, etc, Still I would consider these PL/SQL source files as well.

The other options do indeed include `pls`: 

![image](https://cloud.githubusercontent.com/assets/1747643/25361400/435f96e6-2991-11e7-922f-4e0d2c090013.png)

One final comment I would make is that a series of [example scripts](http://stevenfeuersteinonplsql.blogspot.com.au/2014/06/new-link-to-my-demozip-plsql-script.html) provided by Steven Feurstein, author of the popular Oracle PL/SQL Programming book, mostly use the extension `sql` - one file has the extension `pls`.

Reviewing all the other SQL languages defined in `languages.yml` seem to also put sql at the top of the extension list.

With all that said, [orafaq](http://www.orafaq.com/wiki/File_extensions) does list `pls` under the heading PL/SQL source files, so happy to go with what you had flagged as the primary source file extension.